doc/application/index.rst

```diff
@@ -0,0 +1,12 @@
+.. _application_dev:
+
+Zephyr User Guide
+#################
+
+.. toctree::
+   :maxdepth: 1
+
+   ../porting/porting.rst
+   application.rst
+   ../samples/samples.rst
+   ../reference/kconfig/index.rst
```

Comment:
Instead of here, I could see the Kconfig options moving under the "API" heading, and change that heading to be "APIs and Configuration"  (drop the word "Documentation")